pandas/io/gcs.py

```diff
@@ -0,0 +1,16 @@
+""" GCS support for remote file interactivity """
+try:
+    import gcsfs
+except ImportError:
+    raise ImportError("The gcsfs library is required to handle GCS files")
+
+
+def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
+                           compression=None, mode=None):
+
+    if mode is None:
+        mode = 'rb'
+
+    fs = gcsfs.GCSFileSystem()
+    filepath_or_buffer = fs.open(filepath_or_buffer, mode)
```

Comment:
@bnaul do you think this is an appropriate place to mock? `fs.open` could return a BytesIO object?

Reply:
Added a simple mock-based test here; any other methods I should include besides `read_csv`?